packages/frameworks/vue/src/dialog/dialog-trigger.tsx

```diff
@@ -1,15 +1,24 @@
-import { defineComponent } from 'vue'
+import { computed, defineComponent } from 'vue'
 import { ark, type HTMLArkProps } from '../factory'
+import { usePresenceContext } from '../presence'
 import { useDialogContext } from './dialog-context'
 
 export interface DialogTriggerProps extends HTMLArkProps<'button'> {}
 
 export const DialogTrigger = defineComponent<DialogTriggerProps>(
   (_, { slots, attrs }) => {
     const api = useDialogContext()
+    const presenceApi = usePresenceContext()
+
+    const triggerProps = computed(() => ({
+      ...api.value.triggerProps,
+      'aria-controls': presenceApi.value.isUnmounted
```

Comment:
I don't think this attribute should disappear after the initial rendering of the dialog content with `lazyMount`. Once the dialog content is rendered to the DOM, this attribute should stay attached to the trigger, else the purpose of it is lost.

In other words, the attribute only matters when a user navigates to the trigger, and it should be present if the associated content is visible in the DOM.

I also acknowledge that this attribute [does not have much support with screen readers at the moment](https://a11ysupport.io/tech/aria/aria-controls_attribute); it will not be acknowledged by any of the major SRs, with only JAWS ensuring the user's ability to jump to the controlled element.

My above comments stand in regards to future-proof.

Reply:
Thanks, I've checked the behavior. This attribute disappears only when using `lazyMount` with `unmountOnExit` (like in `Dialog` `LazyMount` story). Otherwise, it stays in place along with lazy-mounted content.

It's the same situation in React, so let's ask Chris.

@cschroeter, what do you think about a11y case that Tyler brought?

Reply:
@TylerAPfledderer @Omikorin 

I just glimpsed over the conversation. `lazyMount` or `unmountOnExit` has often a negative impact on a11y. It is a tradeoff for performance. My recommendation is to avoid using `lazyMount` or `unmountOnExit` if possible.

I've read a bit about this topic here: https://github.com/w3c/aria/issues/995 and as far as I can tell there is no clear winner if `aria-controls` should be removed, empty or just point to the hidden element. 

Under this assumption please keep the current solution as it passes the Storybook a11y Addon and this axe addon for jest some contributor brought up. 

Cheeers

Reply:
Thanks mate. I agree with the current solution.

@TylerAPfledderer I'll dismiss the request then and merge PR. I appreciate yourreview! :)